### PR TITLE
panel meeting search pagination TM-4024

### PR DESCRIPTION
**Pull request by SidelineCory24**
Dual Merge
[API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/762)
[Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/345)

To-do

- [x] add pagination

### CODE REVIEW

**Comment by scott062** [comment]
Seeing weird behavior when change page limits (especially on last page). Toggle filters and page sizes paying attention to last page. Specifically looking for bugs in last page counts etc. You'll see that sometimes the meetings aren't loading in even though it shows counts

**Comment by elizabeth-jimenez** [comment]
> Seeing weird behavior when change page limits (especially on last page). Toggle filters and page sizes paying attention to last page. Specifically looking for bugs in last page counts etc. You'll see that sometimes the meetings aren't loading in even though it shows counts

I'm thinking this may be because of the hackjob pagination we had to do for Panels in order to handle pagination on all the joins

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx [hunk 1]
```diff
@@ -263,6 +280,17 @@ const PanelMeetingSearch = ({ isCDO }) => {
               }
             </div>
         }
+        {
+          !panelMeetingsIsLoading &&
```
Comment (elizabeth-jimenez) [inline src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:284]: why not put it under groupLoading like the other items on the page?

Comment (elizabeth-jimenez) [reply src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:284]: I think with this approach, we run into the possibility of successfully getting the count and showing pagination with our error overlay

Comment (elizabeth-jimenez) [reply src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:284]: let's move this up to below where we rend the panel meeting rows - and update the styling accordingly

**Comment by elizabeth-jimenez** [comment]
Approved pending a bit of feedback and @scott062 reply/verification of https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2470#issuecomment-1448095263

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx [hunk 2]
```diff
@@ -220,31 +226,42 @@ const PanelMeetingSearch = ({ isCDO }) => {
...
-                isLoading={exportIsLoading}
-                disabled={exportDisabled}
+            <div className="panel-results-controls">
+              <SelectForm
+                className="panel-select-box"
+                id="panel-search-results-sort"
+                options={sorts.options}
+                label="Sort by:"
+                defaultSort={ordering}
+                onSelectOption={value => setOrdering(value.target.value)}
```
[the remark this answers is not in the document]
Comment (SidelineCory24) [reply src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:244]: done

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx (hunk 1 above)
Comment (SidelineCory24) [reply src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:284]: moved pagination to below where panel meeting row render

**Comment by SidelineCory24** [comment]
> > Seeing weird behavior when change page limits (especially on last page). Toggle filters and page sizes paying attention to last page. Specifically looking for bugs in last page counts etc. You'll see that sometimes the meetings aren't loading in even though it shows counts
> 
> I'm thinking this may be because of the hackjob pagination we had to do for Panels in order to handle pagination on all the joins

question: is action needed on this, or an expected issue unrelated to this ticket?

**Comment by SidelineCory24** [comment]
> > > Seeing weird behavior when change page limits (especially on last page). Toggle filters and page sizes paying attention to last page. Specifically looking for bugs in last page counts etc. You'll see that sometimes the meetings aren't loading in even though it shows counts
> > 
> > 
> > I'm thinking this may be because of the hackjob pagination we had to do for Panels in order to handle pagination on all the joins
> 
> question: is action needed on this, or is this an expected issue unrelated to this ticket?

double checked fe/mock/insomnia and this is in fact a mock issue

**Comment by SidelineCory24** [comment]
@elizabeth-jimenez @scott062  if there's anything else I need to address let me know, otherwise I think this it gtg. API/Mock have been approved

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx [hunk 3]
```diff
@@ -259,6 +276,16 @@ const PanelMeetingSearch = ({ isCDO }) => {
...
+              {
+                <div className="usa-grid-full react-paginate">
+                  <PaginationWrapper
+                    pageSize={limit}
+                    onPageChange={p => setPage(p.page)}
+                    forcePage={page}
+                    totalResults={count}
+                  />
+                </div>
+              }
```
Comment (elizabeth-jimenez) [inline src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:288]: does it need to be wrapped in {}?

**Review by scott062 — APPROVED** [review]

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx (hunk 3 above)
Comment (SidelineCory24) [reply src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx]: negative, removed 🦅👀

**Review by elizabeth-jimenez — APPROVED** [review]
😸

**Pull request merged**